src/deepsparse/server/server.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from collections import Counter
+from concurrent.futures import ThreadPoolExecutor
+from copy import deepcopy
+from typing import List
+
+import yaml
+
+import uvicorn
+from deepsparse.engine import Context
+from deepsparse.pipeline import Pipeline
+from deepsparse.server.config import (
+    INTEGRATION_LOCAL,
+    INTEGRATION_SAGEMAKER,
+    INTEGRATIONS,
+    EndpointConfig,
+    ServerConfig,
+)
+from fastapi import FastAPI, UploadFile
+from starlette.responses import RedirectResponse
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def start_server(
+    config_path: str, host: str = "0.0.0.0", port: int = 5543, log_level: str = "info"
+):
+    """
+    Starts a FastAPI server with uvicorn with the configuration specified.
+
+    :param config_path: A yaml file with the server config. See :class:`ServerConfig`.
+    :param host: The IP address to bind the server to.
+    :param port: The port to listen on.
+    :param log_level: Log level given to python and uvicorn logging modules.
+    """
+    log_config = deepcopy(uvicorn.config.LOGGING_CONFIG)
+    log_config["loggers"][__name__] = {
+        "handlers": ["default"],
+        "level": log_level.upper(),
+    }
+
+    with open(config_path) as fp:
+        obj = yaml.safe_load(fp)
+    server_config = ServerConfig(**obj)
+    _LOGGER.info(f"Using config: {repr(server_config)}")
+
+    app = _build_app(server_config)
+
+    uvicorn.run(
+        app,
+        host=host,
+        port=port,
+        log_level=log_level,
+        log_config=log_config,
+        # NOTE: only want to have 1 server process so models aren't being copied
```

Comment:
Could you elaborate? What would happen if `workers=2`?

Reply:
num_workers for uvicorn or other server things usually means multiple processes handling requests. In our case this is bad because *I think* each server process would have it's own copy of context/engine. Whereas we want to ensure that there's only 1 context/engine